Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -43,27 +44,53 @@ public FreezeOnMatchCustomization(Type targetType)
         /// that will be satisfied by the frozen specimen.
         /// </param>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="targetType"/> or<paramref name="matcher"/> is null.
+        /// <paramref name="targetType"/> or <paramref name="matcher"/> is null.
         /// </exception>
         public FreezeOnMatchCustomization(
             Type targetType,
             IRequestSpecification matcher)
+            : this(new TypeElement(targetType), matcher)
         {
-            if (targetType == null)
+            this.TargetType = targetType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FreezeOnMatchCustomization"/> class.
+        /// </summary>
+        /// <param name="reflectionElement">
+        /// The <see cref="IReflectionElement"/> used to create specimens to freeze.
+        /// </param>
+        /// <param name="matcher">
+        /// The <see cref="IRequestSpecification"/> used to match the requests
+        /// that will be satisfied by the frozen specimen.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="reflectionElement"/> or <paramref name="matcher"/> is null.
+        /// </exception>
+        public FreezeOnMatchCustomization(
+            IReflectionElement reflectionElement,
+            IRequestSpecification matcher)
+        {
+            if (reflectionElement == null)
             {
-                throw new ArgumentNullException(nameof(targetType));
+                throw new ArgumentNullException(nameof(reflectionElement));
             }
 
             if (matcher == null)
             {
                 throw new ArgumentNullException(nameof(matcher));
             }
 
-            this.TargetType = targetType;
+            this.ReflectionElement = reflectionElement;
             this.Matcher = matcher;
         }
 
         /// <summary>
+        /// The <see cref="ReflectionElement"/> describing the specimen to freeze.
+        /// </summary>
+        public IReflectionElement ReflectionElement { get; }
+
+        /// <summary>
         /// The <see cref="Type"/> of the frozen specimen.
         /// </summary>
         public Type TargetType { get; }
```

Comment:
The property is still valid for other Glue Libraries, so it's too early to make it obsolete.